app/src/sharedTest/java/org/oppia/android/app/player/state/StateFragmentTest.kt

```diff
@@ -546,6 +546,154 @@ class StateFragmentTest {
+    }
+  }
+
+  @Test
+  fun testStateFragment_loadExp_changeConfiguration_thirdState_hasSubmitButton() {
+    launchForExploration(TEST_EXPLORATION_ID_2, shouldSavePartialProgress = false).use {
+      startPlayingExploration()
+      rotateToLandscape()
+
+      playThroughPrototypeState1()
+      playThroughPrototypeState2()
+
+      scrollToViewType(SUBMIT_ANSWER_BUTTON)
+      onView(withId(R.id.submit_answer_button)).check(
+        matches(withText(R.string.state_submit_button))
+      )
+    }
+  }
+
+  @Test
+  fun testStateFragment_loadExp_thirdState_submitAnswer_submitButtonIsEnabled() {
+    launchForExploration(TEST_EXPLORATION_ID_2, shouldSavePartialProgress = false).use {
+      startPlayingExploration()
+      playThroughPrototypeState1()
+      playThroughPrototypeState2()
+
+      selectMultipleChoiceOption(optionPosition = 2, expectedOptionText = "Eagle")
+
+      scrollToViewType(SUBMIT_ANSWER_BUTTON)
+      onView(withId(R.id.submit_answer_button)).check(matches(isEnabled()))
+    }
+  }
+
+  @Test
+  fun testStateFragment_loadExp_thirdState_submitAnswer_clickSubmit_continueButtonIsVisible() {
+    launchForExploration(TEST_EXPLORATION_ID_2, shouldSavePartialProgress = false).use {
+      startPlayingExploration()
+      playThroughPrototypeState1()
+      playThroughPrototypeState2()
+
+      selectMultipleChoiceOption(optionPosition = 2, expectedOptionText = "Eagle")
+      clickSubmitAnswerButton()
+
+      scrollToViewType(CONTINUE_NAVIGATION_BUTTON)
+      onView(withId(R.id.continue_navigation_button)).check(
+        matches(withText(R.string.state_continue_button))
+      )
+    }
+  }
+
+  @Test
+  fun testStateFragment_loadExp_landscape_thirdState_submitAnswer_submitButtonIsEnabled() {
+    launchForExploration(TEST_EXPLORATION_ID_2, shouldSavePartialProgress = false).use {
+      startPlayingExploration()
+      rotateToLandscape()
+      playThroughPrototypeState1()
+      playThroughPrototypeState2()
+
+      selectMultipleChoiceOption(optionPosition = 2, expectedOptionText = "Eagle")
+
+      scrollToViewType(SUBMIT_ANSWER_BUTTON)
+      onView(withId(R.id.submit_answer_button)).check(matches(isEnabled()))
+    }
+  }
+
+  @Test
+  fun testStateFragment_loadExp_land_thirdState_submitAnswer_clickSubmit_continueIsVisible() {
+    launchForExploration(TEST_EXPLORATION_ID_2, shouldSavePartialProgress = false).use {
+      startPlayingExploration()
+      rotateToLandscape()
+      playThroughPrototypeState1()
+      playThroughPrototypeState2()
+
+      selectMultipleChoiceOption(optionPosition = 2, expectedOptionText = "Eagle")
+      clickSubmitAnswerButton()
+
+      scrollToViewType(CONTINUE_NAVIGATION_BUTTON)
+      onView(withId(R.id.continue_navigation_button)).check(
+        matches(withText(R.string.state_continue_button))
+      )
+    }
+  }
+
+  @Test
+  fun testStateFragment_loadExp_thirdState_submitInvalidAnswer_disablesSubmitAndShowsError() {
```

Comment:
This & the next test aren't verifying an error is showing.

Reply:
updated